### DISCUSSION TITLE
[BUG] ¿En la sección user, cómo funciona la jerarquía boss?

### DISCUSSION THREAD

**Discussion by NicolasPLG** [Cotalker/documentation#160] [4. Bug Reports] [answered]
Hola 

Tenemos la siguiente duda relacionado a la jerarquía de jefes. visualmente se ve de la siguiente forma(cotalker)

**1.- Vista trabajador**
![image](https://user-images.githubusercontent.com/61419471/117681656-541c4080-b180-11eb-8ca2-583f8554a07f.png)

**1.1.- Codigo company trabajador**

```
"companies" : [
        {
            "hierarchy" : {
                "boss" : [
                    "5ccb0d4decf6eb5f34614a97"
                ], 
                "peers" : [

                ], 
                "subordinate" : [

                ]
            }, 
            "companyId" : ObjectId("58fe83eda8b7259625b9b15d"), 
            "_id" : ObjectId("59d4f6c1480beb936871c2c3")
        }
    ]
```
**2.- Vista jefe**
![image](https://user-images.githubusercontent.com/61419471/117681698-5c747b80-b180-11eb-9753-561cc7f86dc1.png)
```

**2.1.- Codigo company jefe**
 "companies" : [
        {
            "hierarchy" : {
                "boss" : [
                    ObjectId("5901187fda281cad276780cc")
                ], 
                "peers" : [

                ], 
                "subordinate" : [
                    ObjectId("609942e92f0eac0008ea5389")
                ]
            }, 
            "companyId" : ObjectId("58fe83eda8b7259625b9b15d"), 
            "_id" : ObjectId("5ccb0d4decf6eb5f34614a98")
        }
    ],
```
Comentarios: Sé que existe una diferencia del "hierarchy.boss", uno es id y el otro como objeto, pero visualmente se ve correcto.

Gracias y atentos a sus comentarios

**Comment by rafaelwalter** [comment, marked as answer]
@NicolasPLG Veo que a nivel de datos hay una diferencia, ya que en un caso se guardo como string literal y en otro como `ObjectId`. Lo vamos a corregir.

Esta es la duda?

**Reply by rafaelwalter** [reply]
@NicolasPLG por otro lado, siempre se deberia ver como strings para fuera (atraves de la API), ya que ObjectId es solamente a nivel de base de datos.

**Reply by NicolasPLG** [reply]
esa es una de las dudas, la otra duda, si existen 5 trabajadores con 1 jefe asignado, en la lista del jefe debería aparecer los 5?, como un array de trabajadores?

**Reply by NicolasPLG** [reply]
pero entonces se debería ver los ids para el subordinate del jefe? :>? si es asi llevanto como un tickete o lo tienes registrado? :>

**Reply by rafaelwalter** [reply]
> esa es una de las dudas, la otra duda, si existen 5 trabajadores con 1 jefe asignado, en la lista del jefe debería aparecer los 5?, como un array de trabajadores?

Si! deberian aparecer los 5 trabajadores en el array de subordinates del jefe

**Reply by rafaelwalter** [reply]
> pero entonces se debería ver los ids para el subordinate del jefe? :>? si es asi llevanto como un tickete o lo tienes registrado? :>

Segun el ejemplo que mandaste esta bien!

**Reply by NicolasPLG** [reply]
gracias rafa lo levanto tickete y dejo como solved aqui

Gracias

**Reply by rafaelwalter** [reply]
@NicolasPLG en el siguiente release se vera reflejado un cambio que hicimos para el admin. Ya no se podra configurar el 'boss' en users, solamente los 'subordinates'. Este cambio es por un problema que habia a nivel de datos, justamente el que reportaste aca.

**Reply by rafaelwalter** [reply]
> gracias rafa lo levanto tickete y dejo como solved aqui
> 
> Gracias

no es necesario que lo crees, lo creamos internamente!
